spec/site/command_spec.rb

```diff
@@ -234,10 +234,17 @@
         timeout: SUPERVISOR_CONFIG['status_update_timeout']
       }
       status = "S0096"
+
       expect(result[{"sCI" => status, "n" => "year"}]["s"]).to be == "2020"
       expect(result[{"sCI" => status, "n" => "month"}]["s"]).to be == "9"
       expect(result[{"sCI" => status, "n" => "day"}]["s"]).to be == "29"
       expect(result[{"sCI" => status, "n" => "hour"}]["s"]).to be == "17"
+
+      minutes = (result[{"sCI" => status, "n" => "minute"}]["s"].to_i - 29 + 60) % 60 # Modulo to take care of tests around the shift to a new hour etc.
+      seconds = (result[{"sCI" => status, "n" => "second"}]["s"].to_i - 51 + 60) % 60 # Same as above, but minutes.
+
+      expect(minutes * 60 + seconds).to be <= 300 # Five minutes
```

Comment:
I think it's much easier to use ruby DateTime objects and subtract them. The above is not so easy to follow.
300 seconds can maybe be reduced to something like 30s, what do yu think @otterdahl? It should be a config in e.g. config/ruby.yaml, or we can resue e.g. command_response_timeout or status_response_timeout.

Reply:
You can construct a Time object and then access month, day, etc:
```ruby
      sent = Time.new 2020,9,29,17,29,51,'UTC'
      command_list = build_command_list :M0104, :setDate, {
        securityCode: SECRETS['security_codes'][1],
        year: sent.year,
        month: sent.month,
        day: sent.day,
        minute: sent.min,
        second: sent.sec
      }
```

When you received the status you can do the same thing:

```ruby
      received = Time.new result[{"sCI" => status, "n" => "year"}]["s"],
                          result[{"sCI" => status, "n" => "month"}]["s"],
                          result[{"sCI" => status, "n" => "day"}]["s"],
                          result[{"sCI" => status, "n" => "hour"}]["s"],
                          result[{"sCI" => status, "n" => "minute"}]["s"],
                          result[{"sCI" => status, "n" => "second"}]["s"],
                          'UTC'
```

And then diff them:

```ruby
      diff = received - sent
      expect(diff).to be <= 60
```

Be sure to use UTC timezone throughout.

Reply:
Yeah, I don't know how Ruby too well, but my preferred method of comparing time is to simply convert it to Unix time (epoch) which is just the number of seconds since Jan 01 1970. (UTC). It's then simple to compare the seconds. (It think it's the same a ruby Time object). It doesn't matter that much exactly how far in the future or past you choose to set the clock, as long as you reset it afterwards.

Reply:
i think the current solution is very readable?

Reply:
Yeah. The current solution is better